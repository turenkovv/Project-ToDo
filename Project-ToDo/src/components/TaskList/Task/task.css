.task{
    margin-top: 10px;
    border-radius: 5px;
    background: #393939;
    width: 1190px;
    min-height: 50px;
    height: auto;
    display: flex; 
    flex-direction: row;
    align-items: center;
}
.text{
    color: #FFF;
    font-family: Roboto;
    font-size: 20px;
    font-style: normal;
    font-weight: 400;
    line-height: normal;
}
.name{
    width: 210px;
    height: auto;
    flex-basis: 210px;
    word-break: break-word;
    margin-left: 10px;
}
.desc{
    width: 823px;
    flex-basis: 823px;
    margin-left: 30px;
    margin-right: 20px;
    word-break: break-word;
}
.deleteBtn{
    background-color: transparent;
    border: none;
    margin-left: auto;
    margin-right: 13px;
    
}
.deleteBtn img{
    width: 25px;
}
.checkbox{
    display: block;
    position: relative;
    user-select: none;
    margin-left: 11px;
    width: 30px;
    flex-basis: 36px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
}
.checkbox input{
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    width: 30px;
    height: 30px;
    opacity: 0;
}
.checkmark{
    position: absolute;
    top: -15px;
    left: 0;
    height: 30px;
    width: 30px;
    border-radius: 5px;
    background-color: #FFF;
}
.checkbox:hover input ~ .checkmark{
    background-color: rgb(229, 229, 229);
    transition: 0.2s linear;
}
.checkbox input:checked ~ .checkmark::after{
    display: block;
}
.checkbox .checkmark:after{
    top: 1px;
    left: 2px;
}
.checkmark::after{
    position: absolute;
    display: none;
    content: url("../../../../public/assets/tick.svg");
}

@media screen and (max-width: 1200px) {
    .task{
        width: 100%;
    }
    .text{
        font-size: 17px;
    }
    .name{
        margin-left: 40px;
    }
}