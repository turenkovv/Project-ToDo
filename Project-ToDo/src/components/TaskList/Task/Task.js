import React from 'react';
import { removeFromList, updateStatus } from '../../../store/todoSlice';
import { connect } from 'react-redux';
import './task.css';

const Task = ({ id, name, description, status, removeFromList, updateStatus }) => {
    
    const deleteTask = () => {
        removeFromList({ id, name, description, status });
    }

    const handleCheckbox = () => {
        updateStatus({ id, name, description, status: !status });
    }

    return (
        <div className='task'>
            <label className="checkbox"><input type="checkbox" onChange={handleCheckbox} checked={status}/><span className="checkmark"></span></label>
            <div className="text name">{name}</div>
            <div className="text desc">{description}</div>
            <button className='deleteBtn' onClick={deleteTask}><img src="assets/bin.png" alt=""/></button>
        </div>
    );
}

const mapDispatchToProps = {
    removeFromList, updateStatus
    
}

export default connect(null, mapDispatchToProps)(Task);
