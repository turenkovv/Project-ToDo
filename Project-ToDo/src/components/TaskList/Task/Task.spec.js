import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Task from './Task';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('Task component', () => {
  // Створимо фейковий стан для store
  const initialState = {
    todo: {
      todoList: [
        { id: 1, name: 'Task 1', description: 'Description 1', status: false },
        { id: 2, name: 'Task 2', description: 'Description 2', status: true },
      ],
    },
  };
  const store = mockStore(initialState);// Можете передати початковий стан, якщо потрібно

  test('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Task
          id={1}
          name="Task Name"
          description="Task Description"
          status={false}
          removeFromList={() => {}}
          updateStatus={() => {}}
        />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
