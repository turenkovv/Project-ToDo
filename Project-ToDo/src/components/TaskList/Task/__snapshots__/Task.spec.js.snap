// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component renders correctly 1`] = `
<div
  className="task"
>
  <label
    className="checkbox"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
  <div
    className="text name"
  >
    Task Name
  </div>
  <div
    className="text desc"
  >
    Task Description
  </div>
  <button
    className="deleteBtn"
    onClick={[Function]}
  >
    <img
      alt=""
      src="assets/bin.png"
    />
  </button>
</div>
`;
